attest/attest.go

```diff
@@ -77,6 +78,49 @@ const (
 	StorageKey
 )
 
+type aik interface {
+	Close(*TPM) error
+	Marshal() ([]byte, error)
+	ActivateCredential(tpm *TPM, in EncryptedCredential) ([]byte, error)
+	Quote(t *TPM, nonce []byte, alg tpm2.Algorithm) (*Quote, error)
+	Parameters() AIKParameters
+}
+
+// AIK represents a key which can be used for attestation.
+type AIK struct {
+	aik aik
+}
+
+// Close unloads the AIK from the system.
+func (k *AIK) Close(t *TPM) error {
+	return k.aik.Close(t)
+}
+
+// Marshal encodes the AIK in a format that can be reloaded with tpm.LoadAIK().
+// This method exists to allow consumers to store the key persistently and load
+// it as a later time. Users SHOULD NOT attempt to interpret or extract values
+// from this blob.
+func (k *AIK) Marshal() ([]byte, error) {
+	return k.aik.Marshal()
+}
+
+// ActivateCredential decrypts the specified credential using the key.
+// This operation is synonymous with TPM2_ActivateCredential.
+func (k *AIK) ActivateCredential(tpm *TPM, in EncryptedCredential) ([]byte, error) {
+	return k.aik.ActivateCredential(tpm, in)
+}
+
+// Quote returns a quote over the platform state, signed by the AIK.
+func (k *AIK) Quote(tpm *TPM, nonce []byte, alg tpm2.Algorithm) (*Quote, error) {
```

Comment:
Why is alg `tpm2.Algorithm`? Can't this be a tpm1.2 TPM?

Maybe have this package define its own Algorithm type?